modules/objdetect/src/qrcode.cpp

```diff
@@ -976,23 +986,85 @@ class QRDecode
     size_t getVersion() { return version; }
     std::string getDecodeInformation() { return result_info; }
     bool fullDecodingProcess();
+    bool curvedDecodingProcess();
 protected:
     bool updatePerspective();
     bool versionDefinition();
     bool samplingForVersion();
     bool decodingProcess();
-    Mat original, no_border_intermediate, intermediate, straight;
+    inline double pointPosition(Point2f a, Point2f b , Point2f c);
+    float distancePointToLine(Point2f a, Point2f b , Point2f c);
+    bool getPointsInsideQRCode(vector<Point2f> angle_list);
+    bool computeClosestPoints(vector<Point> result_integer_hull);
+    bool computeSidesPoints(vector<Point> result_integer_hull);
+    vector<Point> getPointsNearUnstablePoint(vector<Point>&side, int start, int end, int step);
+    bool findAndAddStablePoint(vector<Point> result_integer_hull);
+    bool findIndexesCurvedSides();
+    bool findIncompleteIndexesCurvedSides();
+    Mat getMarkersMask();
+    Point findClosestZeroPoint(Point2f original_point);
+    bool findMarkersContours(vector<vector<Point>> &markers_contours);
+    bool findMarkersVerticesPoints(vector<vector<Point>> &markers_edge_points);
+    bool findTempMarkersAddingPoints(vector<std::pair<int,vector<Point>>> &temp_markers_add_points);
+    bool computeMarkersAddingPoints(std::map<int,vector<Point>> &markers_add_points);
+    bool addPointsToSides();
+    vector<vector<float>> spline(vector<int> &x_arr, vector<int> &y_arr);
+    vector<vector<Point2f>> createSpline();
+    vector<vector<Point2f>> divideIntoEvenSegments();
```

Comment:
```suggestion
    vector<vector<float> > computeSpline(vector<float> &x_arr, vector<float> &y_arr);
    vector<vector<Point2f> > createSpline();
    vector<vector<Point2f> > divideIntoEvenSegments();
```
and please replace `computeSpline` outside `QRDecode`